src/debug.cc

```diff
@@ -689,13 +1291,31 @@ Debug::Start(const int section, const int level)
 void
 Debug::Finish()
 {
+    const LoggingSectionGuard sectionGuard;
+
     // TODO: #include "base/CodeContext.h" instead if doing so works well.
     extern std::ostream &CurrentCodeContextDetail(std::ostream &os);
     if (Current->level <= DBG_IMPORTANT)
         Current->buf << CurrentCodeContextDetail;
 
-    // TODO: Optimize to remove at least one extra copy.
-    _db_print(Current->forceAlert, "%s\n", Current->buf.str().c_str());
+    if (Current->waitingForIdle) {
+        const auto past = Current;
+        Current = past->upper;
+        past->upper = nullptr;
+        // do not delete `past` because we store it in WaitingForIdle below
```

Comment:
Please use double-quotes instead of shell-code quotation.
```suggestion
        // do not delete "past" because we store it in WaitingForIdle below
```

Reply:
This is obviously not a shell-related code but a source code comment describing a piece of C++ code. These `backticks` are well-known (in modern markdown and alike) escape sequences that are commonly used to distinguish variable names from words (where the two can be easily confused). Double "quotes" are worse (for this purpose) because, in prose like this, they are commonly used to mean "in indirect sense". I am not forcing others to use backticks, but I hope I will not be forced to use double quotes either.

Reply:
To use yes that is true. But the scripts at times place lines onto stdin/stdout or variables where the shell can get its fingers on the syntax.
If you want to avoid double-quotes it is fine to use singles or brackets etc. But please not the backticks.

Reply:
> the scripts at times place lines onto stdin/stdout or variables where the shell can get its fingers on the syntax

I do not think we should make our code worse because of a possibility that a _buggy_ script will mishandle it. We should be allowed (and are already using) lots of shell-unsafe, HTML-unsafe, whatever-unsafe characters in C++ comments! This is normal/expected -- each syntax has its own set of safe characters.

We should not invent new controversial restrictions like that without strong consensus behind them!

Please withdraw your change request.

>  If you want to avoid double-quotes it is fine to use singles or brackets etc. But please not the backticks.

I do not want to avoid double-quotes. I want to (continue to) use a well-known, modern notation for quoting `code`.

* Single quotes do not emphasize enough that this is a `code` quotation rather than something else. That is especially important for dictionary words like "past".
* Square and angle brackets are commonly used to mean "optional" or "required" in similar contexts.

Please do not misinterpret the bullets above as an invitation to make me evaluate all possible escape sequences. The PR is using the best sequence I know (in this context). If you could not come up with a clearly better one in two attempts, we should stop wasting time on this search.

Reply:
For now, ignoring this. I can patch it later if necessary.